### PR TITLE
Merge 8.4 finalization into trunk

### PR DESCRIPTION
**Pull request by jkmassel**
Merges the `8.4 `release finalization into `trunk`.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt [hunk 1]
```diff
@@ -202,7 +202,11 @@ class MyStoreViewModel @Inject constructor(
             selectedSite.observe()
                 .filter { it?.isJetpackConnected == true }
                 .take(1)
-                .collect { loadStoreStats(activeStatsGranularity.value) }
+                .collect {
+                    activeStatsGranularity.value?.let {
+                        loadStoreStats(it)
+                    }
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: @JorgeMucientes @hichamboushaba is that okay? I believe this value started to be nullable in 09609852557b476d6d43aa9699f872f25bd4a5be commit

**Comment by wzieba** [comment]
@hafizrahman @JorgeMucientes @hichamboushaba hi folks 👋 Could you please help me to resolve failing `Given successful Jetpack installation, When user returns to My Store, Then UI is updated with no JP banner` test?

It was introduced here https://github.com/woocommerce/woocommerce-android/pull/5665 but from briefly look I don't see how to resolve the issue - I'm sure you'll be quicker here with your domain knowledge on this feature.

**Comment by hafizrahman** [comment]
I've added the fix to the unit test, but could use a second pair of eye from @hichamboushaba to check.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt (hunk 1 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt]: Thanks for the heads up @wzieba. An alternative would be to use `_activeStatsGranularity.value` which is not nullable. Wdyt @hichamboushaba ?

```suggestion
                    loadStoreStats(_activeStatsGranularity.value)
```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt [hunk 2]
```diff
@@ -203,9 +203,7 @@ class MyStoreViewModel @Inject constructor(
                 .filter { it?.isJetpackConnected == true }
                 .take(1)
                 .collect {
-                    activeStatsGranularity.value?.let {
-                        loadStoreStats(it)
-                    }
+                    loadStoreStats(_activeStatsGranularity.value)
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt:206]: Thanks!

**Comment by hichamboushaba** [comment]
> I've added the fix to the unit test, but could use a second pair of eye from @hichamboushaba to check.

Thanks for pinging me @hafizrahman, I pushed a second [commit](https://github.com/woocommerce/woocommerce-android/pull/5794/commits/3e0663941c4e524b77b28a93c838c2ca38de48f3) to fix the test without modifying it, since the failure was caused by a change of logic after the merge, it's related to what Wojtek asked about [here](https://github.com/woocommerce/woocommerce-android/pull/5794#discussion_r800213678).

**Explanation of why it started failing:**
One thing that's easy to miss with LiveData is that when it's initialized in a lazy way (using a transformation like `map` or from a `Flow` using `asLiveData`), then it's `value` property won't be computed until it's observed, and since for our test we are not observing it, then it was `null`, and we didn't restart the stats loading here https://github.com/woocommerce/woocommerce-android/blob/f8c8f7f13f8c17e9b8625152d96f81ee59d9f706/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModel.kt#L205-L210

Using the `StateFlow`'s value directly, we will start loading, which fixed the test.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by wzieba** [comment]
@jkmassel 👋 I'm not sure if we or you should merge this but PR is ready to be merged in this form. 

Thanks everyone for quick help on that!

**Review by mokagio — APPROVED** [review]
- [x] `versionName` dropping the `rc` suffix and `versionCode` bump in `build.gradle` 
- [x] `strings.xml` updates
- [x] `release_notes.xml` and `changelogs/*.txt` updates
- [x] Diff from PRs that made it into the `release/` branch since the last beta

**Comment by mokagio** [comment]
I can see @jkmassel was active here not long ago, but I'll merge this to help getting ahead with the code freeze process today.

**Comment by jkmassel** [comment]
Thanks for the quick review @mokagio and for the merge fixes @wzieba, @hafizrahman, and @JorgeMucientes!!

**Pull request merged**